src/common/ipc.ts

```diff
@@ -2,106 +2,46 @@
 // https://www.electronjs.org/docs/api/ipc-main
 // https://www.electronjs.org/docs/api/ipc-renderer
 
-import { ipcMain, ipcRenderer, IpcRendererEvent, WebContents, webContents } from "electron"
+import { ipcMain, ipcRenderer, WebContents, webContents } from "electron"
 import logger from "../main/logger";
-import { getRandId } from "./utils";
 
 export type IpcChannel = string;
 
-export enum IpcMode {
-  SYNC = "sync",
-  ASYNC = "async",
-}
-
-export interface IpcChannelRequest<A extends any[] = any> {
-  msgId: string;
-  args: A;
-}
-
-export interface IpcChannelResponse<T extends any[] = any, E = any> {
-  msgId: string;
-  data?: T;
-  error?: E;
-}
-
 export interface IpcChannelOptions {
   channel: IpcChannel; // main <-> renderer communication channel name
-  mode?: IpcMode; // default: "async", use "sync" as last resort: https://www.electronjs.org/docs/api/ipc-renderer#ipcrenderersendsyncchannel-args
-  handle?: (...args: any[]) => any; // main-process message handler
+  handle?: (...args: any[]) => void; // message handler
```

Comment:
This should probably be `handle?: (...args: any[]) => Promise<void>` since it will be awaited

Reply:
yeah, right